### PR TITLE
WooCommerce: Move wc-api errors to site/status

### PR DESCRIPTION
**Pull request by coderkevin**
This moves the wc-api error state data to underneath state/site/status
for an example of where to keep general top-level status information.

To Test:

1. Run `make test` and ensure all tests pass.

### CODE REVIEW

File: client/extensions/woocommerce/state/site/status/wc-api/error-reducer.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+ */
+import debugFactory from 'debug';
+
+/**
+ * Internal dependencies
+ */
+import { createReducer } from 'state/utils';
+
+import {
+	WOOCOMMERCE_API_SET_ERROR,
```
Comment (coderkevin) [inline client/extensions/woocommerce/state/site/status/wc-api/error-reducer.js]: This error code I think will change soon with the data layer and all, so I'd like to leave it until then.

**Comment by coderkevin** [comment]
@allendav I added selectors in a subsequent PR here for you: #14302  😄

**Comment by coderkevin** [comment]
@DanReyLop Can you please try this out and make sure your shipping zone stuff works? I had to change its reference to the `error` action in the rebase. Thanks!

File: client/extensions/woocommerce/state/site/status/wc-api/reducer.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
+/**
+ * Internal dependencies
```
Comment (DanReyLop) [inline client/extensions/woocommerce/state/site/status/wc-api/reducer.js]: 2 comment blocks called `Internal Dependencies`

Comment (belcherj) [inline client/extensions/woocommerce/state/site/status/wc-api/reducer.js]: @coderkevin to internal blocks

Comment (coderkevin) [inline client/extensions/woocommerce/state/site/status/wc-api/reducer.js]: Got it.

**Comment by ryelle** [comment]
@coderkevin Was this PR also supposed to remove the file `/state/wc-api/error-reducer.js`?

**Comment by coderkevin** [comment]
Oh! Yes, it was. I'm pretty sure I had that in there, but I must have lost it through rebasing. I just created PR #14585 to fix it. Thanks!

**Pull request merged**